python/social_bot/tasks.py

```diff
@@ -388,7 +388,10 @@ def run(self):
                 self._push_reward_queue(max(reward, 0))
                 logging.debug("yielding reward: " + str(reward))
                 agent_sentence = yield TeacherAction(
-                    reward=reward, sentence="well done", done=False)
+                    reward=reward, sentence="well done",
+                    done=not (self._move_goal_during_episode
```

Comment:
should keep compatibility with previous task. Add another flag end_episode_after_successs with default value False.